### ISSUE TITLE
New pytest fixtures not copying test data between runs

### ISSUE THREAD

**Issue by pauleveritt** [sphinx-doc/sphinx#4040]
Subject: Running a pytest fixture should copy the contents of conf.py to temp dir

### Problem

I'm using the `sphinx.testing.fixtures` pytest plugin. I have my `rootdir` setup correctly and my 
`testroot` set correctly.

If I edit the `conf.py` in my testroot and re-run the tests, it isn't copied to the temp directory that is re-used between test runs. I have to set a breakpoint, find the tempdir location, and delete it.

I believe this is the line that is causing the problem:

https://github.com/sphinx-doc/sphinx/blob/e7ffa63463e716382b131da0a9a86beb16ddf804/sphinx/testing/fixtures.py#L71

Is there a flag to get teardown to delete the temp directory? I don't see anything in `fixtures.py` that calls anything to delete the directory.

#### Procedure to reproduce the problem
```
import pytest

pytest_plugins = 'sphinx.testing.fixtures'


@pytest.mark.sphinx('html', testroot='navmenus')
def test_html_entity(app):
    app.build()
    content = (app.outdir / 'index.html').text()
    assert 'Field' in content
```

#### Expected results

Each time I change anything in my testroot, I'd like it to be used as the Sphinx docs source for the next test run.

### Environment info
- OS: macOS 10.12.6
- Python version: 3.6
- Sphinx version: 1.6.3

**Comment by tk0miya** [comment]
Since #5902, we started to follow the custom of pytest. I marked `$SPHINX_TEST_TEMPDIR` as deprecated and started to use pytest's tempdir to the temporary directory on testing by default. It prevents to share the temporary directories between each test invocation. So I think this was already resolved by #5902. Please update your Sphinx to 2.0+ and unset `$SPHINX_TEST_TEMPDIR` if you're using.

Thanks,

**Comment by pauleveritt** [comment]
Thanks!
